js/src/wrappers/ai-sdk.ts

```diff
@@ -25,10 +25,17 @@ import {
  * Wrap an ai-sdk model (created with `.chat()`, `.completion()`, etc.) to add tracing. If Braintrust is
  * not configured, this is a no-op
  *
- * @param model
+ * @param model The AI SDK model to wrap.
+ * @param overrideSpan Optional. A Braintrust Span object to use for tracing this model's operations.
+ *                     If provided, this span will be used instead of creating a new one internally.
+ *                     The lifecycle of this span (e.g., calling .end()) should be managed by the caller
+ *                     if it's passed here.
  * @returns The wrapped object.
  */
-export function wrapAISDKModel<T extends object>(model: T): T {
+export function wrapAISDKModel<T extends object>(
```

Comment:
I know you didn't write this but extending `LanguageModelV1` instead of `object` and removing the generic return type allows us to avoid the `type assertions` in this function:

```suggestion
export function wrapAISDKModel<T extends LanguageModelV1>(model: T) {
  const m = model;
  if (
    m?.specificationVersion === "v1" &&
    typeof m?.provider === "string" &&
    typeof m?.modelId === "string"
  ) {
    return new BraintrustLanguageModelWrapper(m);
  } else {
    console.warn("Unsupported AI SDK model. Not wrapping.");
    return model;
  }
}
```

I haven't tested this so I'm not sure if it breaks anything downstream.

Reply:
yeah, the idea is to change as little as possible to get this into the original repo quicker! My goal is to just add functionality

Reply:
Here is [LanguageModelV1](https://github.com/vercel/ai/blob/main/packages/provider/src/language-model/v1/language-model-v1.ts). I agree, this could be a nice bit a cleanup 🧹